### PR TITLE
Ifc2/lambda integ test fixes

### PR DESCRIPTION
**Pull request by DavidSeptimus-Klotho**
This PR includes a large number of fixes required to get our integ tests working with their base configurations.

Check the individual commit messages for issues addressed in that particular commit.

### CODE REVIEW

File: pkg/infra/iac2/templates/iam_role/factory.ts [hunk 1]
```diff
@@ -13,14 +14,18 @@ interface Args {
...
     return new aws.iam.Role(args.Name, {
         assumeRolePolicy: pulumi.jsonStringify(args.AssumeRolePolicyDoc),
-        //TMPL {{ if .InlinePolicy.Raw }}
-        inlinePolicies: [
-            {
-                name: args.Name,
-                policy: JSON.stringify(args.InlinePolicy),
-            },
+        //TMPL {{- if .InlinePolicies.Raw }}
+        inlinePolicies: args.InlinePolicies,
```
[the remark this answers is not in the document]
Comment (jhsinger-klotho) [reply pkg/infra/iac2/templates/iam_role/factory.ts:18]: nevermind i see below

**Review by jhsinger-klotho — COMMENTED** [review]
some lint and static checks

File: pkg/infra/iac2/templates_compiler.go [hunk 2]
```diff
@@ -435,6 +435,8 @@ func (tc TemplatesCompiler) handleIaCValue(v core.IaCValue, resourceVal *reflect
 		return quoteTsString(secret.SecretName, true), nil
 	case string(core.BUCKET_NAME):
 		return fmt.Sprintf("%s.bucket", tc.getVarName(resource)), nil
+	case string(core.KV_DYNAMODB_TABLE_NAME):
+		return fmt.Sprintf("%s.name", tc.getVarName(resource)), nil
```
[the remark this answers is not in the document]
Comment (DavidSeptimus-Klotho) [reply pkg/infra/iac2/templates_compiler.go:439]: This is an env var that gets converted to an IaCValue, not a standalone IaCValue and so it follows the pattern used by other env vars.

File: pkg/core/resource_graph.go [hunk 3]
```diff
@@ -225,3 +226,27 @@ func (rg *ResourceGraph) getAllDownstreamResourcesSet(source Resource, upstreams
...
+		return zero, err
+	}
+
+	for _, name := range names {
+		if r, ok := rg.GetResource(name).(T); ok {
+			return r, nil
+		}
+	}
+	return zero, fmt.Errorf("resource of type %T not found", zero)
+}
```
[the remark this answers is not in the document]
Comment (DavidSeptimus-Klotho) [reply pkg/core/resource_graph.go]: removed these utilities for now.

File: pkg/infra/iac2/templates_compiler.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jhsinger-klotho) [reply pkg/infra/iac2/templates_compiler.go:439]: i guess im a little confused by that, but definitely not blocking, if its all unnecessary we can remove later

**Review by jhsinger-klotho — APPROVED** [review]

**Pull request merged**